heat/core/sanitation.py

```diff
@@ -381,6 +381,17 @@ def scalar_to_1d(x: DNDarray) -> DNDarray:
     x : DNDarray
         with `x.ndim = 0`
     """
-    return factories.array(
-        x.larray.unsqueeze(0), dtype=x.dtype, split=x.split, comm=x.comm, device=x.device
+    if x.ndim != 0:
+        raise ValueError(
+            "Input needs to be a scalar DNDarray,but was found to be {}d DNDarray".format(x.ndim)
+        )
+    a = x.larray.unsqueeze(0)
+    return DNDarray(
+        a,
+        gshape=tuple(a.size()),
+        dtype=x.dtype,
+        split=x.split,
+        comm=x.comm,
+        device=x.device,
+        balanced=x.is_balanced,
```

Comment:
We can set `gshape = (1,), split=None, balanced=True`, we know it's a 1-element array and scalar DNDarrays are non-distributed by definition.

Reply:
> We can set `gshape = (1,), split=None, balanced=True`, we know it's a 1-element array and scalar DNDarrays are non-distributed by definition.

Yes , got it.